src/cargo/core/workspace.rs

```diff
@@ -1248,13 +1249,29 @@ impl<'cfg> Workspace<'cfg> {
                         all_features: cli_features.all_features,
                         uses_default_features: cli_features.uses_default_features,
                     };
-                    Some((member, feats))
+
+                    // If any member specific features were passed to non-virtual package, it's error
+                    if !member_specific_features.is_empty() {
```

Comment:
I'm not sure if this check is too strict here.

Reply:
Yea, I don't think it can check like this.  It looks like the error with the original code is that it does not check if `member_specific_features` is empty at the end of `members_with_features_old`.